sfge/src/test/java/com/salesforce/rules/unusedmethod/ConstructorsTest.java

```diff
@@ -209,11 +211,54 @@ public void innerConstructorUsedBySiblingInner_expectNoViolation(
             // Declare a method on the sibling inner class that instantiates the tested one.
           + "        /* sfge-disable-stack UnusedMethodRule */\n"
           + "        public void invoker() {\n"
-          + "            " + variable + " obj = new " + constructor + "(false);\n"
+          + "            OuterClass.InnerClass obj = new " + constructor + "(false);\n"
           + "        }\n"
           + "    }\n"
           + "}";
         // spotless:on
         assertNoViolations(sourceCode, 1);
     }
+
+    /**
+     * These tests cover variants of a weird edge case: If an inner class and an outer class share
+     * the same name (e.g., {@code Whatever} and {@code Outer.Whatever}), then calling {@code new
+     * Whatever()} in the inner class's outer/sibling classes should count as an invocation for the
+     * inner class, not the outer class.
+     *
+     * @param arity - The arity of the tested constructor.
+     */
+    @ValueSource(ints = {0, 1})
+    @ParameterizedTest(name = "{displayName}: Constructor of arity {0}")
+    public void externalReferenceSyntaxCollision_expectViolation(Integer arity) {
```

Comment:
It seemed valuable to have tests for this kind of name ambiguity, so I added one.